avahi-autoipd/main.c

```diff
@@ -1218,10 +1236,36 @@ static int loop(int iface, uint32_t addr) {
 
                 if (info.sender_ip_address == addr) {
 
+                    /* Normal conflict? */
                     if (memcmp(hw_address, info.sender_hw_address, ETHER_ADDRLEN)) {
-                        /* Normal conflict */
-                        conflict = 1;
-                        daemon_log(LOG_INFO, "Received conflicting normal ARP packet.");
+                        int should_defend = (state == STATE_RUNNING);
+
+                        /* Have we seen any other conflicting ARP packets within the last DEFEND_INTERVAL seconds? */
+                        if ((should_defend) && (has_timestamp(&last_conflict))) {
+                            /* Check if the time recorded for the previous conflicting ARP packet is recent */
+                            usec = avahi_age(&last_conflict) / 1000ULL;
+                            daemon_log(LOG_DEBUG, "Previous conflict packet received %llu ms  ago", usec);
+                            if (usec > (DEFEND_INTERVAL * 1000)) {
+                                daemon_log(LOG_DEBUG, "Clearing timestamp");
+                                clear_timestamp(&last_conflict);
+                            }
+                        }
+                        if ((should_defend) && (!has_timestamp(&last_conflict))) {
```

Comment:
I believe we are OK. A peer continuously sending conflicting ARP packets outside of the interval (in other words every 10+ seconds) would cause our code to send a single announcement packet in response to each of these (in other words, a single packet every 10+ seconds)